stochastic/continuous/mixed_poisson.py

```diff
@@ -0,0 +1,85 @@
+"""Mixed poisson processes."""
+
+from stochastic.base import Checks
+from stochastic.continuous import PoissonProcess
+
+class MixedPoissonProcess(PoissonProcess):
+    r"""Mixed poisson process. Inherits from the PoissonProcess class.
+
+    A mixed poisson process is a Poisson process for which the rate is a random variate, 
+    a sample taken from a random distribution. On every call of the sample, a new random rate is generated.
+    A Poisson process with rate :math:`\lambda` is a count of occurrences of
+    i.i.d. exponential random variables with mean :math:`1/\lambda`. This class
+    generates samples of times for which cumulative exponential random variables occur. 
+
+    :param function rate_func: random distribution of the rate :math:`\lambda`
+    :param rate_args: arguments to input into the rate_func function
+    :param rate_kwargs: keyword arguments to input into the rate_func function
+    """
+
+    def __init__(self, rate_func,*rate_args,**rate_kwargs):
+        self.rate_func = rate_func
+        self.rate_args = rate_args
+        self.rate_kwargs = rate_kwargs
+        self.gen_rate()
+
+    def __str__(self):
+        return "Mixed Poisson process with rate {r}.".format(r=str(self.rate))
+
+    def __repr__(self):
+        return "MixedPoissonProcess(rate={r})".format(r=str(self.rate))
+
+    @property
+    def rate_func(self):
+        """Current rate's random distribution."""
+        return self._rate_func
+        
+    @rate_func.setter
+    def rate_func(self, value):
+        self._rate_func = value
+        self.gen_rate()
+        
+    @property
+    def rate_kwargs(self):
+        """Keyword arguments for rate generation using given random distribution."""
+        return self._rate_kwargs
+        
+    @rate_kwargs.setter
+    def rate_kwargs(self, value):
+        self._rate_kwargs = value
+        self.gen_rate()
+        
+    @property
+    def rate_args(self):
+        """Arguments for rate generation using given random distribution."""
+        return self._rate_args
+        
+    @rate_args.setter
+    def rate_args(self, value):
+        self._rate_args = value
+        self.gen_rate()
+        
+    @property
+    def rate(self):
+        """Current rate."""
+        return self._rate
+        
+    @rate.setter
+    def rate(self, value):
+        rate_func, rate_args,rate_kwargs = value 
+        self._rate_func=rate_func
+        self._rate_args=rate_args
+        self._rate_kwargs=rate_kwargs
```

Comment:
The rate setter and getter here is not needed. On sample you should just be doing self.rate = self.gen_rate() and gen_rate should return the new sample. Just use the parent rate getter/setter.